hazelcast/src/main/java/com/hazelcast/internal/util/OsHelper.java

```diff
@@ -46,4 +46,13 @@ public static boolean isLinux() {
     public static boolean isUnixFamily() {
         return (OS.contains("nix") || OS.contains("nux") || OS.contains("aix"));
     }
+
+    /**
+     * Returns {@code true} if the system is a Mac OS.
+     *
+     * @return {@code true} if the current system is Mac.
+     */
+    public static boolean isMac() {
+        return (OS.contains("mac") || OS.contains("darwin"));
```

Comment:
maybe we should also search for string "osx" for some macOS versions: https://support.apple.com/en-us/HT201260 (see the table)

Reply:
tested `os.name` property on macos 11.4 & 10.13 and both return `Mac OS X`

Reply:
>In 2012, with the release of OS X 10.8 Mountain Lion, the name of the system was shortened from Mac OS X to OS X

>In 2016, with the release of macOS 10.12 Sierra, the name was changed from OS X to macOS

Based on this [wikipedia](https://en.wikipedia.org/wiki/MacOS#OS_X) entry, only checking for "mac" is not sufficient.

Reply:
I think Apple's OS names don't have much to do with what the JDK returns for `os.name` property. The [JDK 8 sources since 2013](http://hg.openjdk.java.net/jdk8/jdk8/jdk/file/1a3de3cdc684/src/solaris/native/java/lang/java_props_macosx.c#l132) hard-code the os name to `Mac OS X`. This seems to be the same as [current JDK status](https://github.com/openjdk/jdk/blob/739769c8fc4b496f08a92225a12d07414537b6c0/src/java.base/macosx/native/libjava/java_props_macosx.c#L233-L235). Also, JDK sources themselves rely on matching `Mac OS X` like [here](https://github.com/openjdk/jdk/blob/739769c8fc4b496f08a92225a12d07414537b6c0/test/jdk/java/lang/ProcessBuilder/DestroyTest.java#L115) and [here](https://github.com/openjdk/jdk/blob/739769c8fc4b496f08a92225a12d07414537b6c0/test/jdk/java/lang/ProcessBuilder/DestroyTest.java#L115).

Reply:
Thanks then it seems there is no missing case.